services/src/handlers/layers.rs

```diff
@@ -455,3 +464,391 @@ async fn layer_handler<C: Context>(
+    pub layer_collection_id: LayerCollectionId,
+}
+
+/// Add a new collection to an existing collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layers/collections",
+    request_body = AddCollectionRequest,
+    responses(
+        (status = 200, description = "OK", body = AddCollectionResponse,
+            example = json!({
+                "layerCollectionId": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_collection<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add collections to their stuff
+    ctx: web::Data<C>,
+    request: web::Json<AddCollectionRequest>,
+) -> Result<web::Json<AddCollectionResponse>> {
+    let request = request.into_inner();
+
+    let parent_id = request.parent_id;
+    let add_collection = request.layer_collection.validated()?;
+
+    let layer_collection_id = ctx
+        .layer_db_ref()
+        .add_collection(add_collection, &parent_id)
+        .await?;
+
+    Ok(web::Json(AddCollectionResponse {
+        layer_collection_id,
+    }))
+}
+
+#[derive(Debug, Serialize, Deserialize, IntoParams)]
+struct RemoveCollectionFilter {
+    /// Specify this layer filter to only remove a specifc layer from a collection.
+    layer: Option<LayerId>,
+}
+
+/// Remove an entire collection or a layer from a collection
+#[utoipa::path(
+    tag = "Layers",
+    delete,
+    path = "/layers/collections/{collection}",
+    params(
+        ("collection" = LayerCollectionId, description = "Layer collection id"),
+        RemoveCollectionFilter,
+    ),
+    responses(
+        (status = 200, description = "OK")
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn remove_collection<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to remove their collections
+    ctx: web::Data<C>,
+    collection: web::Path<LayerCollectionId>,
+    filter: web::Query<RemoveCollectionFilter>,
+) -> Result<HttpResponse> {
+    if let Some(layer_id) = filter.layer.as_ref() {
+        ctx.layer_db_ref()
+            .remove_layer_from_collection(layer_id, &collection)
+            .await?;
+    } else {
+        ctx.layer_db_ref().remove_collection(&collection).await?;
+    }
+
+    Ok(HttpResponse::Ok().finish())
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{
+        contexts::{InMemoryContext, Session},
+        util::tests::send_test_request,
+        workflows::workflow::Workflow,
+    };
+    use actix_web::{http::header, test};
+    use actix_web_httpauth::headers::authorization::Bearer;
+    use geoengine_datatypes::util::test::TestDefault;
+    use geoengine_operators::{
+        engine::VectorOperator,
+        mock::{MockPointSource, MockPointSourceParams},
+    };
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_add_layer_to_collection() {
+        // fixing this avoids random admin session tokens
```

Comment:
why would it be random? It's set in the Settings-test.toml. Please remove this or it might interfere with other tests that use the admin_session_token and run concurrently.